src/interface/swarm_container.cpp

```diff
@@ -0,0 +1,104 @@
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#include "swarm_container.hpp"
+#include "bvals/cc/bvals_cc.hpp"
+#include "globals.hpp" // my_rank
+#include "mesh/mesh.hpp"
+#include <cstdlib>
+#include <memory>
+#include <utility>
+#include <vector>
+
+namespace parthenon {
+
+void SwarmContainer::Add(const std::vector<std::string> labelArray,
+                         const Metadata &metadata) {
+  // generate the vector and call Add
+  for (auto label : labelArray) {
+    Add(label, metadata);
+  }
+}
+
+///
+/// The internal routine for allocating a particle swarm.  This subroutine
+/// is topology aware and will allocate accordingly.
+///
+/// @param label the name of the variable
+/// @param metadata the metadata associated with the particle
+void SwarmContainer::Add(const std::string label, const Metadata &metadata) {
+  printf("Adding swarm to SwarmContainer!\n");
+  if (swarmMap_.find(label) != swarmMap_.end()) {
+    throw std::invalid_argument("swarm " + label + " already enrolled during Add()!");
+  }
+
+  printf("about to make shared!");
```

Comment:
Are these leftover debugging comments that should be removed?

Reply:
Yes sorry, I forgot to clean up this file. Fixed.